Updated the generate responses on the question show/ index view.

Hellrungj opened the pull request:
the response text links are now changed to basic text and updated to generate one after other after on a different line.

Commit:
Updated the responses text link to basic text and update the show view to genrate the responses one after other after on a different line

hstoebel commented on app/views/questions/show.html.haml:
```diff
@@ -29,11 +29,12 @@
-				= link_to response.text, response
+				= response.text
 			%td
 				= response.user_id
```
instead of displaying the user's id, could we show the user's name?

jstoebel commented:
This looks great! I just have one suggestion: instead of displaying the user's id, could we show their user name?

Commit:
Make Major change to the show, index and edit view along with there layout surounding those views.

Commit:
This save file (_navbar.html.haml.save) should not be here.

This code useful and needs to be deleted.

Commit:
This save file (_navbar.html.haml.save) should not be here.

This code useless and we don't a save file and current code.

Commit:
This save file (_navbar.html.haml.save) should not be here.

This code is useless. The project don't need save files.

Commit:
Updated the views to reflected both current views and mine own. This way both views are updated and look like the rest of the codeworkout site.

Commit:
Merge branch 'hellrungj-UI-testing' of https://github.com/jstoebel/code-workout into hellrungj-UI-testing
Updating the repo.

Commit:
Made changes so the the buttons could work. Now, all the button work. Along with the buttons, I add a breadcrumb to the new view.

Commit:
Update the show view to allow more space for the question body and add the index view to the new and add view breadcumb

Commit:
Updated the file for the pull request.

Hellrungj commented:
Fixed the conflicts.

hstoebel commented on app/views/layouts/_breadcrumb.html.haml:
```diff
@@ -58,5 +58,29 @@
     - else
       %li.active= @organization.abbreviation
 
+  - elsif @edit
```
I wonder if there is a more general way to tell the user what part of the Q&A controller they are in. maybe look in params to see what controller, action and id is being requested.

hstoebel commented on app/views/questions/index.html.haml:
```diff
@@ -1,26 +1,30 @@
 
 -if @questions.size == 0
-	%em = No Questions Found
+  %em = No Questions Found
```
If this isn't indented won't it run regardless of @questions.size

hstoebel commented on app/views/questions/show.html.haml:
```diff
@@ -1,90 +1,91 @@
-                %td= link_to "Edit", edit_question_path(@question), class: "btn btn-info"
-                -#%input.btn-info.btn-info{:type => "submit", :value => "Delete"}/
+%h3
+  = "Exercise: #{@question.exercise_id}"
```
could we see the name of the exercise rather than its id `@question.exercise.name`

Hellrungj replied:
I tried to change the line to @question.exercise.name and @question.exercise_id.name both broke. I am not how to show the exercise name.

Commit:
adjustments to abilities
removed object level authorization for new as it was rejecting auth for records with no user_id. This ability is already checked at the action level. Record level auth isn't needed as currently everyone has the right to Question#new
removed universal permission to create questions (the question must belong to you to create it

Commit:
Update the index view to only allow one new question button and modified the _breadcrumb file so the edit view show the question that it is editing.

Commit:
Merge branch 'hellrungj-UI-testing' of https://github.com/jstoebel/code-workout into hellrungj-UI-testing

Hellrungj commented:
I ready again to pull. I tried working on use the params method  of integrating the _breadcrumb but it didn't work at all. Plus, all information I research does really support the params method.

jstoebel commented:
cool! Let's review in class tomorrow!

Commit:
Add the search bar to the indew and show view.

Commit:
Move the _breadcrumb code to the question view in the show action to reduce the dry code.

Commit:
Fixed a bug with the edit view. The Edit view was showing it id rather than its title.

Commit:
Updated the show view and ran gem bundle install which modified the schema.rb file.

Commit:
Fixed the conflict with the schema.rb

Commit:
Updated the show view
Deleted the HEAD tag of the top of the file.

Commit:
Merge branch 'master' of https://github.com/jstoebel/code-workout into hellrungj-UI-testing

Commit:
Modifed the show view

Commit:
Updated the show view and migrate files
Updated the show fixing the conflects of the prevous pull
Updated the migrate file bug to allow the rails to run - Dropped the drop_up_votes migrate file

Commit:
Updated the repo
Updated the index view to fix any conflicts
Updated the question controler to fix any conflicts
Updated the show vbiew to fix any conflicts
Edit the show view's table to produce a better page
Run the mirgate which update the schema.rd file

hstoebel commented on app/controllers/questions_controller.rb:
```diff
@@ -31,6 +31,18 @@ def show
     authorize! :read, @question
     @responses = Response.all.where(question_id: params[:id])
     @response = Response.new
+    @title = ''
```
consider the `truncate` helper to simplify this. http://api.rubyonrails.org/classes/String.html#method-i-truncate

hstoebel commented on app/views/questions/show.html.haml:
```diff
@@ -1,92 +1,90 @@
-                    %br/
-                    %br/
+%h3
+  = "User: #{@question.user_id}"
```
get the user's name: `"#{@question.user.first_name} #{@question.user.last_name}"`

Commit:
Updated the show view and question controller
Show View: Now, shows the users name for the question and the responses
Question View: Simplied the breadcrumb by adding the method truncate
